facenet/fr_utils.py

```diff
@@ -0,0 +1,201 @@
+  'inception_4a_pool_conv', 'inception_4a_pool_bn',
+  'inception_4a_1x1_conv', 'inception_4a_1x1_bn',
+  'inception_4e_3x3_conv1', 'inception_4e_3x3_conv2', 'inception_4e_3x3_bn1', 'inception_4e_3x3_bn2',
+  'inception_4e_5x5_conv1', 'inception_4e_5x5_conv2', 'inception_4e_5x5_bn1', 'inception_4e_5x5_bn2',
+  'inception_5a_3x3_conv1', 'inception_5a_3x3_conv2', 'inception_5a_3x3_bn1', 'inception_5a_3x3_bn2',
+  'inception_5a_pool_conv', 'inception_5a_pool_bn',
+  'inception_5a_1x1_conv', 'inception_5a_1x1_bn',
+  'inception_5b_3x3_conv1', 'inception_5b_3x3_conv2', 'inception_5b_3x3_bn1', 'inception_5b_3x3_bn2',
+  'inception_5b_pool_conv', 'inception_5b_pool_bn',
+  'inception_5b_1x1_conv', 'inception_5b_1x1_bn',
+  'dense_layer'
+]
+
+conv_shape = {
+  'conv1': [64, 3, 7, 7],
+  'conv2': [64, 64, 1, 1],
+  'conv3': [192, 64, 3, 3],
+  'inception_3a_1x1_conv': [64, 192, 1, 1],
+  'inception_3a_pool_conv': [32, 192, 1, 1],
+  'inception_3a_5x5_conv1': [16, 192, 1, 1],
+  'inception_3a_5x5_conv2': [32, 16, 5, 5],
+  'inception_3a_3x3_conv1': [96, 192, 1, 1],
+  'inception_3a_3x3_conv2': [128, 96, 3, 3],
+  'inception_3b_3x3_conv1': [96, 256, 1, 1],
+  'inception_3b_3x3_conv2': [128, 96, 3, 3],
+  'inception_3b_5x5_conv1': [32, 256, 1, 1],
+  'inception_3b_5x5_conv2': [64, 32, 5, 5],
+  'inception_3b_pool_conv': [64, 256, 1, 1],
+  'inception_3b_1x1_conv': [64, 256, 1, 1],
+  'inception_3c_3x3_conv1': [128, 320, 1, 1],
+  'inception_3c_3x3_conv2': [256, 128, 3, 3],
+  'inception_3c_5x5_conv1': [32, 320, 1, 1],
+  'inception_3c_5x5_conv2': [64, 32, 5, 5],
+  'inception_4a_3x3_conv1': [96, 640, 1, 1],
+  'inception_4a_3x3_conv2': [192, 96, 3, 3],
+  'inception_4a_5x5_conv1': [32, 640, 1, 1,],
+  'inception_4a_5x5_conv2': [64, 32, 5, 5],
+  'inception_4a_pool_conv': [128, 640, 1, 1],
+  'inception_4a_1x1_conv': [256, 640, 1, 1],
+  'inception_4e_3x3_conv1': [160, 640, 1, 1],
+  'inception_4e_3x3_conv2': [256, 160, 3, 3],
+  'inception_4e_5x5_conv1': [64, 640, 1, 1],
+  'inception_4e_5x5_conv2': [128, 64, 5, 5],
+  'inception_5a_3x3_conv1': [96, 1024, 1, 1],
+  'inception_5a_3x3_conv2': [384, 96, 3, 3],
+  'inception_5a_pool_conv': [96, 1024, 1, 1],
+  'inception_5a_1x1_conv': [256, 1024, 1, 1],
+  'inception_5b_3x3_conv1': [96, 736, 1, 1],
+  'inception_5b_3x3_conv2': [384, 96, 3, 3],
+  'inception_5b_pool_conv': [96, 736, 1, 1],
+  'inception_5b_1x1_conv': [256, 736, 1, 1],
+}
+
+def load_weights_from_FaceNet(FRmodel):
+    # Load weights from csv files (which was exported from Openface torch model)
+    weights = WEIGHTS
+    weights_dict = load_weights()
+
+    # Set layer weights of the model
+    for name in weights:
+        if FRmodel.get_layer(name) != None:
+            FRmodel.get_layer(name).set_weights(weights_dict[name])
+        elif model.get_layer(name) != None:
+            model.get_layer(name).set_weights(weights_dict[name])
+
+def load_weights():
+    # Set weights path
+    dirPath = './weights'
```

Comment:
please delete the weights that are stored in this directory, lets keep them in GCS and make a local copy when we run this file. Add the weights directory to .gitignore

Reply:
Yes I will do that